src/Microdown-DocumentBrowser/MicPharoPackageCommentResourceReference.class.st

```diff
@@ -0,0 +1,208 @@
+"
+My prime purpose it to allow access to the package comment.
+
+My uri syntax is: `comment://package/categoryName`.
+
+## loadMicrodown
+My `loadMicrodown` returns the package comment of the package named categoryName or empty text if categoryName is not a package or if the package has no comment. Notice, `loadMicrodown` will return the synthesized comment (created by the class side method `buildMicroDownUsing:withComment:`)
+
+When categoryName is a package, my two methods `contents` and `contents:` reads and writes the contents of the package comment.
+
+## loadDirectory
+My 'loadDirectory' has been designed to provide a deep directory. 
+
+Most class category names are of the form 'aaa-bbb-ccc-ddd'. A call to `loadDirectory` with categoryName being 'aaa' will give children 'aaa-xxx','aaa-yyy','aaa-zzz'. A call with 'aaa-bbb' will return children 'aaa-bbb-xxx', 'aaa-bbb-yyy' etc.
+
+### Details
+In `comment://package/categoryName`, categoryName can be one of three cases:
+* A package name, with children being the tags or classes if the package has no tags
+* A tag of the form 'PackageName-tag', with children being the classes of the tag
+* A package name prefix of the form 'AI-Algorithms' wich is neither a package or a tag in a package, with children being those categories of which 'AI-Algorithms' is a prefix.
+
+A special case is somewhat common, for example 'Microdown' is both a package name and a prefix for other packages.
+In this case, the prefix 'Microdown' takes precedence  over package, and `comment://package/Microdown` has as children those categories prefixed my 'Microdown' _and_ the Microdown package.
+
+"
+Class {
+	#name : #MicPharoPackageCommentResourceReference,
+	#superclass : #MicPharoCommentResourceReference,
+	#instVars : [
+		'kind',
+		'categoryName'
+	],
+	#category : #'Microdown-DocumentBrowser-ResourceModel'
+}
+
+{ #category : #private }
+MicPharoPackageCommentResourceReference >> browserIcon [
```

Comment:
Hi kasper 

I would package browser Icon in the doc browser package to avoid ui in the core. 

S

Reply:
The class `MicPharoPackageCommentResourceReference` is actually a model class in the browser package, not the core. The browser package adds `browserIcon` method to some of the core classes, but as extension methods.